superset/connectors/base/models.py

```diff
@@ -319,8 +320,13 @@ def data_for_slices(  # pylint: disable=too-many-locals
                 if "column" in filter_config
             )
 
+            # for legacy dashboard imports which have the wrong query_context in them
+            try:
+                query_context = slc.get_query_context()
+            except DatasetNotFoundError:
+                query_context = None
```

Comment:
@AAfghahi and I also talked about maybe writing a script/migration as a follow up that will remove any existing query_context fields from charts if they contain a dataset id. Either way, it seems that if query context is failing in any way here, that it would be just as fine not to use it.